client/extensions/woocommerce/components/query-locations/index.js

```diff
@@ -0,0 +1,65 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchLocations } from 'woocommerce/state/sites/locations/actions';
+import { areLocationsLoaded } from 'woocommerce/state/sites/locations/selectors';
+
+class QueryLocations extends Component {
+	static propTypes = {
+		fetchLocations: PropTypes.func,
+		loaded: PropTypes.bool,
+		siteId: PropTypes.number.isRequired,
+	};
+
+	componentWillMount = () => {
+		const { siteId, loaded } = this.props;
+
+		if ( siteId && ! loaded ) {
+			this.props.fetchLocations( siteId );
+		}
+	};
+
+	componentWillReceiveProps = ( { siteId, loaded } ) => {
+		if ( ! siteId ) {
+			return;
+		}
+
+		if ( siteId !== this.props.siteId && ! loaded ) {
+			this.props.fetchLocations( siteId );
+		}
+	};
+
+	render = () => {
+		return null;
+	};
+}
+
+const mapStateToProps = ( state, ownProps ) => {
+	const loaded = areLocationsLoaded( state, ownProps.siteId );
+
+	return {
+		loaded,
+	};
+};
+
+const mapDispatchToProps = dispatch => {
+	return bindActionCreators(
+		{
+			fetchLocations,
+		},
+		dispatch
+	);
```

Comment:
If you aren't doing anything fancy, use the object shorthand:
```jsx
const mapDispatchToProps = { fetchLocations };
```

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/21297/commits/698f0228879b86ac921f12fd74df554610db7238